cmd/kusk/cmd/deploy.go

```diff
@@ -206,9 +207,24 @@ var deployCmd = &cobra.Command{
 func getParsedAndValidatedOpenAPISpec(apiSpecPath string) (string, error) {
 	const KuskExtensionKey = "x-kusk"
 
-	parsedApiSpec, err := spec.NewParser(&openapi3.Loader{IsExternalRefsAllowed: true}).Parse(apiSpecPath)
-	if err != nil {
-		return "", err
+	parsedApiSpec := &openapi3.T{}
+
+	overlay, err := overlays.NewOverlay(apiSpecPath)
+	if err == nil {
+		overlayPath, err := overlay.Apply()
+		if err != nil {
+			return "", err
+		}
+
+		parsedApiSpec, err = spec.NewParser(&openapi3.Loader{IsExternalRefsAllowed: true}).Parse(overlayPath)
```

Comment:
`spec.NewParser(&openapi3.Loader{IsExternalRefsAllowed: true}).Parse` has an identical line below bar the parameter. Perhaps use:

```go
func NewParser(path string) &openapi3.T{} {
  return spec.NewParser(&openapi3.Loader{IsExternalRefsAllowed: true}).Parse(path)
}
```

Reply:
Or just create a parser further up
```
parser := spec.NewParser(&openapi3.Loader{IsExternalRefsAllowed: true})
...
parser.Parse(...) // where needed
```

Reply:
Yeh, even better :+1:.